DEP 11 -- Add accessibility team draft DEP (#69)
django/deps: draft/0011-accessibility-team.rst

==========================
DEP 11: Accessibility Team
==========================

:DEP: 11
:Author: Tom Carrick
:Implementation Team: Tom Carrick, Thibaud Colas, others to be determined.
:Shepherd: Carlton Gibson
:Status: Draft
:Type: Process
:Created: 2020-06-29
:Last-Modified: 2020-06-29

.. contents:: Table of Contents
  :depth: 3
  :local:

Abstract
========

This DEP proposes the formation of an Accessibility Team to encourage projects

> maintained by the Django Software Foundation to be accessible to as diverse a
> group as possible, particularly people with disabilities that make using the

Reviewer: ```suggestion
maintained by the Django Software Foundation to be accessible to as many
people as possible, particularly those with disabilities that make using the
```

web more difficult.

Specification
=============

The Accessibility Team (informally, a11y team) shall be formed.

Membership

> ----------

Reviewer: Unsure if this needs spelling out, but presumably the team will be listed on https://www.djangoproject.com/foundation/teams/  .

Author: I'm also unsure. It seems a bit unnecessary and I also couldn't think of a good way to word it into the DEP.

Reviewer: Fair.

> The team shall not have a fixed size, but instead will grow and shrink

Reviewer: Should we spell out in this section that Tom and Thibaud are the first members? Are there any other first members?

Author: I don't feel any particular need. I don't see a reason why initial members should be prioritised over anyone else, and we're in the implementation team anyway.

organically as members choose to leave, and when new members are deemed to be
required by the rest of the team. Membership of the team works similarly to the
`Code of Conduct Committee <https://github.com/django/code-of-conduct/blob/master/membership.md>`_.
New members shall be chosen from a list of volunteers, or if there is a lack
of volunteers, an advertisement will be published on the Django website.
Priority will be given to volunteers who, in no particular order:

- Have disabilities that make using the web and/or web development more
  difficult.
- Have relevant, positive contributions or expertise in the field.
- Have a record of contributing to Django.

Members shall remain in the team for a fixed-term of 9 months, after which

> they must opt-in to remain on the team. Membership will also be terminated by:

Reviewer: Will opting-in renew for another 9 months? Will team members need to continue opting-in each 9 months in perpetuity?

Author: Yes. I've clarified the wording a bit here.

- Becoming disqualified due to actions taken by the Code of Conduct committee
  of the Django Software Foundation.

- A vote of the Technical Board, or full consensus of the rest of the
  Accessibility Team, if the team is considered too large, the person is not

>   making positive contributions, or any other sound reason.

Reviewer 2: Is it worth explicitly stating that they can also terminate their own membership?

Author: I think it is. I've added it.

Responsibilities
----------------

The remit of the Accessibility Team shall include all user-facing components
of projects maintained by the Django Software Foundation. This includes but is
not limited to:

- User-facing parts of Django, such as default HTML output in forms and such,
  and the Django admin and its default theme, HTML, and UI components.

- All websites and other software projects maintained by the DSF, such as the

>   Django website, including documentation, Django Snippets, the issue tracker,
>   etc.

Reviewer: I think the team will cover *all* documentation right, no just that on the Django website - e.g. Channels on Read the Docs ?

```suggestion
  Django website, projects' documentation, Django Snippets, the issue tracker, etc.
```

The responsibilities of the Accessibility Team are expected to change over
time, and are to be decided by consensus of the team, with input from the
Technical Board as required. To begin, several areas have been identified:

- Deciding on any relevant accessibility guidelines to follow, such as WCAG,
  and at which conformance level.

- Implementing automated testing to catch issues, working with the ops

>   team as needed to integrate this into the CI process.

Reviewer: nitpick - though I guess most work will be on Django core, multiple projects = multiple CI proceses
```suggestion
  team as needed to integrate this into CI processes.
```

- Coordinating regular manual accessibility audits on all relevant projects.

- Coordinating the fixing of accessibility issues and the improvement of the
  accessibility in general in Django and associated projects.

- Writing and maintaining documentation relating to accessibility, such as
  a statement of commitment to accessibility issues, and contribution
  guidelines.

> - Reviewing accessibility fixes, improvements and other tickets that may affect

Reviewer 2: I think adding an explicit goal of identifying areas where the framework could provide API improvements/new features that help users of Django also create accessible sites would be good?

This is kind of covered by this statement, but rather implicitly

Author: Yes, that's a good point, and I've added it.

>   the accessibility of the project.

Reviewer: ```suggestion
  accessibility of any relevant project.
```

To aid in reviewing, the Accessibility Team members shall be added to a team
in the GitHub organization with read access to relevant repositories, so that
they may be requested to review pull requests, at the discretion of the author,

> reviewer, or other party.

Reviewer: Can we specify the name of that github team here as (`accessibility`?), so anyone reading the DEP can see the name to request on their PR's without having to hunt elsewhere?

Author: Done.

While this is mostly noted in the real world, the web also experiences this
effect. For example, keyboard shortcuts are used to allow keyboard-only users
navigate a site more efficiently, but this positive benefit affects everyone,
not just the physically disabled.

The Django admin currently exhibits many accessibility issues, such as forms
fields missing labels, low color contrast, and a lacklustre keyboard-only
experience among other issues. Other parts of the project have problems such
as non-semantic HTML generated by forms. Futher problems exist on other sites
maintained by the DSF, such as the
`Django website <https://www.djangoproject.com/>`__,
`Django Snippets <https://djangosnippets.org/>`_, and others.

Rationale
=========

An alternative is to go on as usual, leaving accessibility in the hands of
individual contributors when writing and reviewing code. However, given the
accessibility issues that exist in the Django admin and on the
`Django website <https://www.djangoproject.com/>`__, this doesn't seem to be
enough.

Another option is to implement some basic standards, such as conforming to WCAG

> and setting up a CI to test. This approach is better than nothing but it

Reviewer: ```suggestion
and setting up a CI tests. This approach is better than nothing but it
```
